apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/securerandom/SecureRandom.kt

```diff
@@ -19,13 +20,21 @@ actual class SecureRandom actual constructor(
      */
     override fun nextBytes(size: Int): ByteArray {
         val arr = Uint8Array(size)
-        return crypto.getRandomValues(arr).buffer.toByteArray()
+        return if (isNode) {
+            _require("crypto").randomFillSync(arr)
+        } else {
+            global.crypto.getRandomValues(arr)
+        }
     }
 
     actual companion object : SecureRandomStaticInterface {
         override fun generateSeed(numBytes: Int): ByteArray {
             val arr = Uint8Array(numBytes)
-            return crypto.getRandomValues(arr).buffer.toByteArray()
+            return if (isNode) {
+                _require("crypto").randomFillSync(arr)
+            } else {
+                global.crypto.getRandomValues(arr)
+            }
```

Comment:
We can't because they are very different. The `crypto` module is different for the `Node.js` and `Browser`. Both have other usage interfaces.

If we want/need to implement it like you are suggesting. We would create our interface and class that implements that interface, and in each method internally, we would have the if condition based on the environment.